sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -2362,81 +2355,29 @@ String generateKeyId(String keyName, String keyVersion) {
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<KeyRotationPolicy>> updateKeyRotationPolicyWithResponse(String name, KeyRotationPolicyProperties keyRotationPolicyProperties) {
+    public Mono<Response<KeyRotationPolicy>> updateKeyRotationPolicyWithResponse(String name, KeyRotationPolicy keyRotationPolicy) {
         try {
             return withContext(context ->
-                updateKeyRotationPolicyWithResponse(name, keyRotationPolicyProperties, context));
+                updateKeyRotationPolicyWithResponse(name, keyRotationPolicy, context));
         } catch (RuntimeException e) {
             return monoError(logger, e);
         }
     }
 
-    Mono<Response<KeyRotationPolicy>> updateKeyRotationPolicyWithResponse(String name, KeyRotationPolicyProperties keyRotationPolicyProperties, Context context) {
+    Mono<Response<KeyRotationPolicy>> updateKeyRotationPolicyWithResponse(String name, KeyRotationPolicy keyRotationPolicy, Context context) {
         try {
             if (CoreUtils.isNullOrEmpty(name)) {
                 return monoError(logger, new IllegalArgumentException("'name' cannot be null or empty"));
             }
 
-            List<LifetimeAction> lifetimeActions = new ArrayList<>();
-
-            for (KeyRotationLifetimeAction lifetimeAction : keyRotationPolicyProperties.getLifetimeActions()) {
-                lifetimeActions.add(new LifetimeAction()
-                    .setAction(new LifetimeActionsType()
-                        .setType(lifetimeAction.getType()))
-                    .setTrigger(new LifetimeActionTrigger()
-                        .setTimeAfterCreate(lifetimeAction.getTimeAfterCreate())
-                        .setTimeBeforeExpiry(lifetimeAction.getTimeBeforeExpiry())));
-            }
-
-            com.azure.security.keyvault.keys.implementation.models.KeyRotationPolicy keyRotationPolicy =
-                new com.azure.security.keyvault.keys.implementation.models.KeyRotationPolicy()
-                    .setAttributes(new KeyRotationPolicyAttributes()
-                        .setExpiryTime(keyRotationPolicyProperties.getExpiryTime()))
-                    .setLifetimeActions(lifetimeActions);
-
             return service.updateKeyRotationPolicy(vaultUrl, name, keyServiceVersion.getVersion(), keyRotationPolicy,
                     "application/json", context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
                 .doOnRequest(ignored -> logger.verbose("Updating key rotation policy for key with name.", name))
                 .doOnSuccess(response -> logger.verbose("Updated key rotation policy for key with name.", name))
-                .doOnError(error -> logger.warning("Failed to retrieve key rotation policy - {}", error))
-                .map(response -> new SimpleResponse<>(response, toKeyRotationPolicy(response.getValue())));
+                .doOnError(error -> logger.warning("Failed to retrieve key rotation policy - {}", error));
```

Comment:
Unrelated to this PR but we should start using the supplier methods in the logger to avoid creating strings when verbose logs are disabled.

Reply:
I agree - even though we aren't in a critical perf path we should always do the small things that prevent it from ever being an issue.